core/templates/pages/exploration-editor-page/history-tab/services/compare-versions.service.spec.ts

```diff
@@ -428,217 +451,180 @@ describe('Compare versions service', function() {
     }];
 
     // Tests for getDiffGraphData on linear commits
-    it('should detect changed, renamed and added states', function() {
-      $httpBackend.expect('GET', '/explorehandler/init/0?v=1')
```

Comment:
Such tests have to be replaced where a http request is made to a service that has been upgraded to Angular. The upgraded service now uses `HttpClient` to issue http request and not `$http`. `$httpBackend` does not work with `HttpClient`, it is the `HttpTestingController` that comes into play then (used few lines below).

So, what I do is the following:
1. Migrate the service and it's spec file.
2. Add the migrated service to UpgradedServices.ts
3. Run the frontend tests and see what all tests fail
4. These all failing tests are the ones where $httpBackend is used and should be ideally replaced with httpTestingController
5. Fix all such tests

This is not increasing any of our efforts since the end state of the project would be no `$httpBackend` but all `httpTestingController`.

Reply:
Is this worth documenting on the wiki page?